### PR TITLE
Add missing null reference checks

### PR DESCRIPTION
**Pull request by pascalberger**
Add missing null reference checks and make sure an `ArgumentNullException` is thrown in all cases

### CODE REVIEW

File: src/Cake.Git/GitBranch.cs [hunk 1]
```diff
@@ -49,6 +50,11 @@ public sealed class GitBranch
         /// <param name="repository">The repository.</param>
         public GitBranch(Repository repository)
         {
+            if (repository == null)
```
Comment (soroshsabz) [inline src/Cake.Git/GitBranch.cs]: If you use C# 7 and earlier you can use `_ = repository ?? throw new ArgumentNullException`

https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/operators/null-coalescing-operator

File: src/Cake.Git/GitBranch.cs [hunk 2]
```diff
@@ -64,6 +70,16 @@ public GitBranch(Repository repository)
         /// <param name="branch">The branch.</param>
         internal GitBranch(Repository repository, Branch branch)
         {
+            if (repository == null)
```
Comment (soroshsabz) [inline src/Cake.Git/GitBranch.cs]: Same as above

File: src/Cake.Git/GitCommit.cs [hunk 3]
```diff
@@ -53,7 +53,7 @@ internal GitCommit(Commit commit)
         {
             if (commit == null)
             {
-                throw new ArgumentException("Source commit can't be null.", nameof(commit));
+                throw new ArgumentNullException(nameof(commit));
```
Comment (soroshsabz) [inline src/Cake.Git/GitCommit.cs]: Same as above

File: src/Cake.Git/GitDiffFile.cs [hunk 4]
```diff
@@ -35,6 +36,11 @@ public class GitDiffFile
 
         internal GitDiffFile(TreeEntryChanges change)
         {
+            if (change == null)
```
Comment (soroshsabz) [inline src/Cake.Git/GitDiffFile.cs]: Same as above

**Review by devlead — APPROVED** [review]
LGTM 👍

**Comment by devlead** [comment]
@pascalberger your changes have been merged, thanks for your contribution 👍

**Pull request merged**